lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -72,46 +82,76 @@ def fragment_cache(cached_hash, non_cached_hash)
           ActiveModel::Serializer::Adapter::JsonApi::FragmentCache.new.fragment_cache(root, cached_hash, non_cached_hash)
         end
 
-        protected
+        private
 
-        attr_reader :fieldset
+        # Build the requested resource objects.
+        # @return [Array] [primary, included] Pair of arrays containing primary and included
+        #   resources objects respectively.
+        #
+        # @api private
+        def resource_objects_for(serializer)
+          resources = _resource_objects_for(serializer, @include_tree, true).values
+          resources.each_with_object([[], []]) do |resource, (primary, included)|
+            is_primary = resource.delete(:is_primary)
+            if is_primary
+              primary.push(resource)
+            else
+              included.push(resource)
+            end
+          end
+        end
 
-        private
+        # Recursively build all requested resource objects.
+        # @return [Hash<ResourceIdentifier, ResourceObject>] Hash of resource objects keyed by resource identifiers.
```

Comment:
yay!

sidenote - are ResourceIdentifiers and ResourceObjects actual things in AMS?